test/utility_test.cc

```diff
@@ -98,10 +98,14 @@ class UtilityAddressResolutionTest : public TestWithParam<Envoy::Network::Addres
     envoy::config::core::v3::TypedExtensionConfig typed_dns_resolver_config;
     Envoy::Network::DnsResolverFactory& dns_resolver_factory =
         Envoy::Network::createDefaultDnsResolverFactory(typed_dns_resolver_config);
-    Envoy::Network::DnsResolverSharedPtr dns_resolver =
+    absl::StatusOr<Envoy::Network::DnsResolverSharedPtr> dns_resolver =
         dns_resolver_factory.createDnsResolver(*dispatcher, *api, typed_dns_resolver_config);
+    if (!dns_resolver.ok()) {
+      throw Envoy::EnvoyException(
```

Comment:
Actually, I think the tests expect the nighthawk source code to throw exceptions. For example: https://github.com/envoyproxy/nighthawk/blob/main/source/common/uri_impl.cc#L31

So I think we will have to keep the EXPECT_THROW calls in the tests. (Unless we alter the source API).

I didn't know how to deal with the createDnsResolver failure case with the given test infrastructure, so I just threw an exception. I don't think this particular test file tests createDnsResolver cases (although other unit tests I added in process_test do).

Reply:
Thank you for explaining. We should definitely keep the EXPECT_THROW if that is the case. Throwing when dns_resolver is fairly reasonable. However, I think the most dogmatic way to ensure the precondition is met though is by asserting.

Reply:
I changed it here: https://github.com/envoyproxy/nighthawk/pull/1180/commits/2c97645a86b9a05bb5aef9b0ea2b70da5f4020c4

I changed one instance of this exception throwing. It won't let me change the one in the class method. It won't build, and I think we shouldn't modify the test api to return a statusor since the source api throws exceptions, which our test case expects.